src/io/flutter/actions/DeviceSelectorAction.java

```diff
@@ -71,8 +69,9 @@ public void update(final AnActionEvent e) {
 
     final FlutterDaemonService service = FlutterDaemonService.getInstance(project);
 
-    if (!isListening) {
-      isListening = true;
+    if (!knownProjects.contains(project)) {
+      knownProjects.add(project);
+      Disposer.register(project, () -> knownProjects.remove(project));
```

Comment:
Down the road it'd be great to migrate this logic into `FlutterDaemonService` where we've already got disposers registered on projects but that makes sense probably for a bigger cleanup.